tests/unit/projection-test.js

```diff
@@ -0,0 +1,1464 @@
+import { module, test, skip } from 'qunit';
+import { setupTest }  from 'ember-qunit';
+import Ember from 'ember';
+import MegamorphicModel from 'ember-m3/model';
+import SchemaManager from 'ember-m3/schema-manager';
+import { initialize as initializeStore } from 'ember-m3/initializers/m3-store';
+import { watchProperties } from '../helpers/watch-property';
+
+const {
+  get,
+  set,
+  run,
+  RSVP: { Promise },
+} = Ember;
+
+/*
+  Non-normalized ClassPaths are used for the data returned by the adapter and pushed to the store
+  but for everything post-entry we use the normalized version.
+ */
+const BOOK_CLASS_PATH = 'com.example.bookstore.Book';
+const NORM_BOOK_CLASS_PATH = 'com.example.bookstore.book';
+const BOOK_EXCERPT_PROJECTION_CLASS_PATH = 'com.example.bookstore.projection.BookExcerpt';
+const NORM_BOOK_EXCERPT_PROJECTION_CLASS_PATH = 'com.example.bookstore.projection.book-excerpt';
+const BOOK_PREVIEW_PROJECTION_CLASS_PATH = 'com.example.bookstore.projection.BookPreview';
+const NORM_BOOK_PREVIEW_PROJECTION_CLASS_PATH = 'com.example.bookstore.projection.book-preview';
+// never used since this is embedded
+// const PROJECTED_AUTHOR_CLASS = 'com.example.bookstore.projectedType.projectedAuthor';
+const NORM_PROJECTED_AUTHOR_CLASS = 'com.example.bookstore.projected-type.projected-author';
+const PUBLISHER_CLASS = 'com.example.bookstore.publisher';
+const PROJECTED_PUBLISHER_CLASS = 'com.example.bookstore.projectedType.projectedPublisher';
+const NORM_PROJECTED_PUBLISHER_CLASS = 'com.example.bookstore.projected-type.projected-publisher';
+
+module('unit/projection', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    initializeStore(this);
+
+    this.store = function() {
+      return this.owner.lookup('service:store');
+    };
```

Comment:
since we're using `setupTest` there's no reason to use the old-style `let`; we may as well `this.store = this.owner.lookup('service:store')`

Reply:
👍